core/src/main/java/com/airbnb/aerosolve/core/transforms/MoveFloatToStringTransform.java

```diff
@@ -48,17 +52,37 @@ public void doTransform(FeatureVector featureVector) {
     Map<String, Set<String>> stringFeatures = featureVector.getStringFeatures();
     Set<String> output = Util.getOrCreateStringFeature(outputName, stringFeatures);
 
-    for (String key : keys) {
-      if (feature1.containsKey(key)) {
-        Double dbl = feature1.get(key);
-        if (dbl > cap) {
-          dbl = cap;
-        }
-
-        Double quantized = LinearLogQuantizeTransform.quantize(dbl, bucket);
-        output.add(key + '=' + quantized);
+    if (keys != null) {
+      for (String key : keys) {
+        moveFloat(feature1, output, key, cap, bucket);
         feature1.remove(key);
       }
+    } else {
+      for (Iterator<Entry<String, Double>> iterator = feature1.entrySet().iterator();
+          iterator.hasNext();) {
+        Entry<String, Double> entry = iterator.next();
+        String key = entry.getKey();
+
+        moveFloat(feature1, output, key, cap, bucket);
+        iterator.remove();
+      }
+    }
+  }
+
+  public static void moveFloat(
+      Map<String, Double> feature1,
+      Set<String> output,
+      String key,
+      double cap,
+      double bucket) {
+    if (feature1.containsKey(key)) {
+      Double dbl = feature1.get(key);
+      if (dbl > cap) {
+        dbl = cap;
+      }
+
+      Double quantized = LinearLogQuantizeTransform.quantize(dbl, bucket);
```

Comment:
we discussed about how confusing this is to have quantize in `LinearLogQuantizeTransform`, perhaps you can refactor this part in a future PR